src/components/general/DatePicker.js

```diff
@@ -15,99 +16,76 @@ class DatePicker extends Component {
 			month: date.length === 3 ? date[1] : "",
 			day: date.length === 3 ? date[2] : "",
 			year: date.length === 3 ? date[0] : "",
+			focused: date.length === 3,
 			monthArr: Array.from({ length: 12 }, (v, k) => k + 1),
-			dayArr: Array.from({ length: 31 }, (v, k) => k + 1),
-			yearArr: Array.from({ length: 101 }, (v, k) => new Date().getFullYear() - k),
-			focused: date.length === 3
+			yearArr: Array.from({ length: 20 }, (v, k) => new Date().getFullYear() + k) // We only need about 20 years ahead... right?
 		};
 	}
 
-		static propTypes = {
-			value: PropTypes.object,
-			placeholder: PropTypes.string.isRequired,
-			onSelect: PropTypes.func.isRequired
-		}
+	static propTypes = {
+		value: PropTypes.object,
+		placeholder: PropTypes.string.isRequired,
+		onSelect: PropTypes.func.isRequired
+	}
 
-		prepend0(item) {
-			if (item < 10)
-				return "0" + item;
+	// Returns an array of days in the current state's month when called
+	dayArr() {
+		const {
+			month,
+			day,
+			year
+		} = this.state;
 
-			return item;
-		}
+		// Return the correct array of days based on the selected month
+		// Defining # of days for each month is easier than whatever monstrosity was here before
```

Comment:
Remove unnecessary comments.
The return should be mentioned in the JSDocs above.